domain/modeldefaults/state/state.go

```diff
@@ -230,10 +235,276 @@ WHERE m.uuid = ?
 	}, nil
 }
 
-// NewState returns a new State for interacting with the underlying model
-// defaults.
-func NewState(factory database.TxnRunnerFactory) *State {
-	return &State{
-		StateBase: domain.NewStateBase(factory),
+// GetModelCloudDetails returns the cloud UUID and region for the given model.
+// If the model is not found, an error specifying [modelerrors,NotFound] is returned.
+func (s *State) GetModelCloudDetails(ctx context.Context, uuid coremodel.UUID) (cloud.UUID, string, error) {
```

Comment:
Why return the cloud as a uuid and the region as a name rather than both names or uuids?

Same question for many of the args and return values in this state package.

Reply:
The return values fit the use case. The existing facade callers and upstream consumers expect region name. The cloud UUID is used by subsequent service logic.